### PR TITLE
New service deploy option: interval

### LINKED ISSUE #656: Erosion resistance
https://devcenter.heroku.com/articles/erosion-resistance

### PR DESCRIPTION
**Pull request by jakolehm**
Fixes #656

### CODE REVIEW

**Comment by jakolehm** [comment]
I don't really like name of this feature/option (`redeploy_after`). Could we come up with something better?

/cc @nevalla @miskun @matti

File: cli/lib/kontena/cli/services/services_helper.rb [hunk 1]
```diff
@@ -319,6 +319,20 @@ def parse_secrets(secret_opts)
...
+        # @return [Integer]
+        def parse_relative_time(time)
+          if time.end_with?('min')
+            time.to_i * 60
+          elsif time.end_with?('h')
+            time.to_i * 60 * 60
+          elsif time.end_with?('d')
+            time.to_i * 60 * 60 * 24
+          else
+            memory.to_i
```
Comment (nevalla) [inline cli/lib/kontena/cli/services/services_helper.rb]: should be time.to_i

**Comment by nevalla** [comment]
How about redeploy_interval or auto_deploy_interval?

File: cli/lib/kontena/cli/services/services_helper.rb (hunk 1 above)
Comment (jakolehm) [inline cli/lib/kontena/cli/services/services_helper.rb]: Good catch, was going to write tests for this PR :)

**Comment by jakolehm** [comment]
Interval makes sense, maybe something like this:

```
deploy:
  interval: 8h
```

**Comment by nevalla** [comment]
I like it. Simple and clear.

**Comment by christofferh** [comment]
Isn't the semantics more of `restart` rather than `redeploy`?

**Comment by jakolehm** [comment]
@christofferh, restart does not help with "erosion" so much because it just restarts the actual container. Deploy will pull latest version of image and swap container to fresh one.

**Comment by christofferh** [comment]
@jakolehm, IMO this should be an idempotent action. Automatically pulling for the latest version is something I would consider dangerous and not part of 'erosion resistance'.

**Comment by jakolehm** [comment]
Part of erosion resistance is that app/service should receive security updates. This can't be done without pulling new version of defined "image:tag". 

If this is not acceptable I think we should introduce separate option for image pulls (always, on_demand).

**Comment by christofferh** [comment]
That's true if you're not in complete control of the underlying OS of the service. e.g. the case of Heroku.

Yes, it should to be optional with image pulls if `redeploy` is the way you guys wanna go. But I just don't think it's the right way to go 😢 

(You guys are awesome! 💯 )

**Comment by matti** [comment]
This is a nice feature for somebody who is adventurous, but we really need the other feature that @christofferh talked about.

We want to test every patch version that goes out: when you are using ruby and node and other LATEST stuff by the LATEST people, it's very common that patch versions break stuff.

So another issue with title of "New service restart option: interval"

```
restart:
  interval: 2h
```

**Comment by matti** [comment]
(the word `interval` is great here)

**Comment by jakolehm** [comment]
Just a clarification: this will only pull fresh image of what user has defined in kontena.yml (or service definition). 

So user still has total control of versions that will be deployed. It's no different what already happens when service instance moves to new node (by some event that is not controlled by user, for example node reboot).

**Pull request merged**